app/lib/pdf_filler/md502_pdf.rb

```diff
@@ -0,0 +1,58 @@
+module PdfFiller
+  class Md502Pdf
+    include PdfHelper
+
+    def source_pdf_name
+      "md502-TY2023"
+    end
+
+    def initialize(submission)
+      @submission = submission
+
+      # Most PDF fields are grabbed right off the XML
+      builder = StateFile::StateInformationService.submission_builder_class(:md)
+      @xml_document = builder.new(submission).document
+    end
+
+    def hash_for_pdf
+      {
+        "Enter day and month of Fiscal Year beginning": formatted_date(@xml_document.at('ReturnHeaderState TaxPeriodBeginDt')&.text, "%m-%d"),
+        "Enter day and month of Fiscal Year Ending": formatted_date(@xml_document.at('ReturnHeaderState TaxPeriodEndDt')&.text, "%m-%d"),
+        "Enter social security number": @xml_document.at('Primary TaxpayerSSN')&.text,
+        'Enter spouse\'s social security number': @xml_document.at('Secondary TaxpayerSSN')&.text,
+        "Enter your first name": @xml_document.at('Primary TaxpayerName FirstName')&.text,
+        "Enter your middle initial": @xml_document.at('Primary TaxpayerName MiddleInitial')&.text,
+        "Enter your last name": @xml_document.at('Primary TaxpayerName LastName')&.text,
+        'Enter Spouse\'s First Name': @xml_document.at('Secondary TaxpayerName FirstName')&.text,
+        'Enter Spouse\'s middle initial': @xml_document.at('Secondary TaxpayerName MiddleInitial')&.text,
+        'Enter Spouse\'s last name': @xml_document.at('Secondary TaxpayerName LastName')&.text,
+        "Check Box - 1": filing_status(:filing_status_single?) ? 'Yes' : 'Off',
+        "Check Box - 2": filing_status(:filing_status_mfj?) ? 'Yes' : 'Off',
+        "Check Box - 3": filing_status(:filing_status_mfs?) ? 'No' : 'Off', # For some reason, "Yes" does not check the box for MFS
+        "MARRIED FILING Enter spouse's social security number": spouse_ssn_if_mfs,
+        "Check Box - 4": filing_status(:filing_status_hoh?) ? 'Yes' : 'Off',
+        "Check Box - 5": filing_status(:filing_status_qw?) ? 'Yes' : 'Off',
+        "6. Check here": claimed_as_dependent? ? 'Yes' : 'Off',
+        "Text Box 96": @xml_document.at('ReturnHeaderState Filer Primary USPhone')&.text,
+      }
+    end
+
+    def spouse_ssn_if_mfs
+      filing_status(:filing_status_mfs?) ? @xml_document.at('MFSSpouseSSN')&.text : nil
+    end
+
+    def claimed_as_dependent?
+      @submission.data_source.direct_file_data.claimed_as_dependent?
+    end
+
+    def filing_status(method)
+      claimed_as_dependent? ? false : @submission.data_source.send(method)
+    end
+
+    def formatted_date(date_str, format)
+      return if date_str.nil?
+
+      Date.parse(date_str)&.strftime(format)
+    end
```

Comment:
[PEBBLE] could we add this to `PdfHelper` since we have the same thing in `nc_d400_pdf.rb`?